framework/src/state_machine/state_machine.ts

```diff
@@ -211,6 +211,11 @@ export class StateMachine {
 	}
 
 	public async verifyAssets(ctx: BlockContext): Promise<void> {
+		for (const asset of ctx.assets.getAll()) {
+			if (this._findModule(asset.module) === undefined) {
+				throw new Error(`Module ${asset.module} is not registered.`);
+			}
+		}
```

Comment:
asset existence check is done in verify asset instead of module because ABI `init` is removed